content/docs/tasks/traffic-management/egress-gateway/index.md

```diff
@@ -515,23 +537,70 @@ The output should be the same as in the previous section.
     apiVersion: networking.istio.io/v1alpha3
     kind: DestinationRule
     metadata:
-      name: set-sni-for-egress-gateway
+      name: egressgateway-for-cnn
     spec:
       host: istio-egressgateway.istio-system.svc.cluster.local
-      trafficPolicy:
-        loadBalancer:
-          simple: ROUND_ROBIN
-        portLevelSettings:
-        - port:
-            number: 443
-          tls:
-            mode: MUTUAL
-            clientCertificate: /etc/certs/cert-chain.pem
-            privateKey: /etc/certs/key.pem
-            caCertificates: /etc/certs/root-cert.pem
-            subjectAltNames:
-            - spiffe://cluster.local/ns/istio-system/sa/istio-egressgateway-service-account
-            sni: edition.cnn.com
+      subsets:
+      - name: cnn
+        trafficPolicy:
+          loadBalancer:
+            simple: ROUND_ROBIN
+          portLevelSettings:
+          - port:
+              number: 443
+            tls:
+              mode: MUTUAL
+              clientCertificate: /etc/certs/cert-chain.pem
+              privateKey: /etc/certs/key.pem
+              caCertificates: /etc/certs/root-cert.pem
+              subjectAltNames:
+              - spiffe://cluster.local/ns/istio-system/sa/istio-egressgateway-service-account
+              sni: edition.cnn.com
+    ---
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: direct-cnn-through-egress-gateway
+    spec:
+      hosts:
+      - edition.cnn.com
+      gateways:
+      - mesh
+      tls:
+      - match:
+        - gateways:
+          - mesh
+          port: 443
+          sni_hosts:
+          - edition.cnn.com
+        route:
+        - destination:
+            host: istio-egressgateway.istio-system.svc.cluster.local
+            subset: cnn
+            port:
+              number: 443
+          weight: 100
+    ---
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: cnn
+    spec:
+      hosts:
+      - edition.cnn.com
+      gateways:
+      - istio-egressgateway
+      tcp:
+      - match:
+        - gateways:
+          - istio-egressgateway
+          port: 443
+        route:
+        - destination:
+            host: edition.cnn.com
+            port:
+              number: 443
+          weight: 100
```

Comment:
I suggest not to split into two virtual services. Because it gives the false impression that you can have multiple virtual services for a host. Instead, combine both into a single virtual service

Reply:
Note https://github.com/istio/istio/issues/7361#issuecomment-407612066. This was the solution for https://github.com/istio/istio/issues/7361, See  `3` (not working) vs `3a` (working) in https://github.com/istio/istio/issues/7361#issuecomment-407720960.

Reply:
@rshriram Done, updated https://github.com/istio/istio/issues/7361#issuecomment-409141493 also.